Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1

```diff
@@ -250,50 +250,58 @@ Set-Alias -Scope Global Rebuild-Solution Invoke-RebuildSolution
-            "`"/nr:$noReuse`"",
-            "`"/p:Platform=$platformEnvironmentVariable`""
-        )
-
-        if (-not [string]::IsNullOrWhiteSpace($maxCpuCount)) {
-            $msBuildParameters += "`"/m:$maxCpuCount`""
-        }
-        else {
-            $msBuildParameters += "`"/m`""
-        }
-
         $params = @{
             Path                           = $solutionPath
-            MsBuildParameters              = ($msBuildParameters -join ' ')
-            ShowBuildOutputInCurrentWindow = $true
+            BuildConfiguration             = $buildConfiguration
             LogVerbosityLevel              = $verbosity
         }
 
-        if (-not [string]::IsNullOrWhiteSpace($msBuildFilePath)) { $params.MsBuildFilePath = $msBuildFilePath }
-
         ($params).GetEnumerator() | Sort-Object -Property Name | Format-Table -HideTableHeaders -AutoSize -Wrap | Out-Host
-        $buildResult = Invoke-MsBuild @params
 
-        if ($null -eq $buildResult.BuildSucceeded) { throw "Unsure if build passed or failed: $($buildResult.Message)" }
-        if ($buildResult.BuildSucceeded -eq $true) { return }
-        if ($buildResult.BuildSucceeded -eq $false) {
+		# Local Variables.
+		$BuildLogDirectoryPath = $env:Temp
+		
+		# Always get the full path to the Log files directory.
+		$BuildLogDirectoryPath = [System.IO.Path]::GetFullPath($BuildLogDirectoryPath)
+		
+		# Local Variables.
+		$solutionFileName = (Get-ItemProperty -LiteralPath $solutionPath).Name
+		$buildLogFilePath = (Join-Path -Path $BuildLogDirectoryPath -ChildPath $solutionFileName) + ".msbuild.log"
+		$buildErrorsLogFilePath = (Join-Path -Path $BuildLogDirectoryPath -ChildPath $solutionFileName) + ".msbuild.errors.log"
+		$buildSucceeded = $null
+        
+		# Build
+		dotnet build $solutionPath -c $buildConfiguration -v $verbosity /flp:v=$verbosity /flp:logfile=$buildLogFilePath
+		
+		# If we can't find the build's log file in order to inspect it, write a warning and return null.
+		if (!(Test-Path -LiteralPath $buildLogFilePath -PathType Leaf))
+		{
+			$buildSucceeded = $null
+			
+			Write-Warning ("Cannot find the build log file at '$buildLogFilePath', so unable to determine if build succeeded or not.")
+		}
+		
+		# Get if the build succeeded or not.
+		[bool] $buildOutputDoesNotContainFailureMessage = $null -eq (Select-String -Path $buildLogFilePath -Pattern "Build FAILED." -SimpleMatch)
+		$buildSucceeded = $buildOutputDoesNotContainFailureMessage
+
+        if ($null -eq $buildSucceeded) { throw "Unsure if build passed or failed: FAILED to build ""$solutionPath"". Please check the build log ""$buildLogFilePath"" for details." }
+        if ($buildSucceeded -eq $true) 
+		{ 
+			if (Test-Path -LiteralPath $buildLogFilePath -PathType Leaf) { Remove-Item -LiteralPath $buildLogFilePath -Force }
+			if (Test-Path -LiteralPath $buildErrorsLogFilePath -PathType Leaf) { Remove-Item -LiteralPath $buildErrorsLogFilePath -Force }
+			return
+		}
```

Comment:
Might be nice to keep the logs around especially if it just overwrites by default.

Reply:
personally i do not like the log files being placed in appdata, should we consider using %temp% instead? (thinking long term with linux or macos build)

Reply:
Repo root might be nice too. Would need to add it to .gitignore.

Reply:
Currently we are using %temp% folder